SampleOrigin/wwwroot/Index.html

```diff
@@ -0,0 +1,49 @@
+﻿<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8" />
+    <script type="text/javascript">
+        
+        function createCORSRequest(method, url) {          
+            var request = new XMLHttpRequest();
+            request.open(method, url, true);
+            return request;
+        }
+
+        // Make the CORS request.
+        function makeCORSRequest(method) {
+
+            // Destination server with CORS enabled.
+            var url = 'http://destination.sample.com/api';
+            var request = createCORSRequest(method , url);
+
+            if (!request) {
+                alert('CORS not supported');
+                return;
+            }
+            
+            // Response handlers.
+            request.onload = function () {
+                var text = request.responseText;
+                alert('Response from CORS '+method+' request to ' + url + ': ' + text);
+            };
+
+            request.onerror = function () {
+                alert('There was an error making the request for method ' + method);
+            };
+
+            request.send();
+        }
+
+    </script>
+    <title></title>
+</head>
+<body>
+    <form method="post">
+            <input id="clickMe" type="button" value="CORS Request" onclick="makeCORSRequest('GET');" />
```

Comment:
Maybe keep it "somewhat" static but have 3 - 4 common scenarios? Or if it's easy enough to make it dynamic, that's fine too: just have textboxes with default values and a "Make request" button that uses the values. That would be neat.